pages/auth/login.page.js

```diff
@@ -0,0 +1,245 @@
+import dynamic from "next/dynamic";
+import Head from "next/head";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styles from "../../styles/form.module.css";
+const Header = dynamic(() => import("../../components/layout/Header"));
+const Footer = dynamic(() => import("../../components/layout/Footer"));
+
+import Visibillity from "../../public/assets/Visibillity.jsx";
+import VisibillityOff from "../../public/assets/VisibillityOff.jsx";
+
+import { useAuth } from "../../context/AuthContext";
+import { encryptData, decryptData } from "../../hook/encryptDecrypt.js";
+
+function login() {
+  const router = useRouter();
+  const [error, setError] = useState("");
+  const [entityType, setEntityType] = useState("user");
+
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+
+  const [formData, setFormData] = useState({
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const {
+    isMentorLoggedIn,
+    setIsMentorLoggedIn,
+    isUserLoggedIn,
+    setIsUserLoggedIn,
+  } = useAuth();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const togglePasswordVisibility = () => {
+    setIsPasswordVisible((prevState) => !prevState);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setError("");
+    if (formData.password !== formData.confirmPassword) {
+      return toast.error("Password does not match!");
+    }
+    console.log(formData);
+    try {
+      const url = `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/auth?type=${entityType}`;
+      const { data: res } = await axios.post(url, formData);
+      let entityData = {};
+      if (entityType === "user") {
+        entityData = {
+          token: res.data,
+          user_name: res.data.fullName,
+          user_email: res.data.email,
+          user_id: res.data.id,
+        };
+        localStorage.setItem("userData", encryptData(entityData));
+        setIsUserLoggedIn(true);
+      } else if (entityType === "mentor") {
+        entityData = {
+          mentor_name: res.fullName,
+          mentorToken: res.mentorToken,
+        };
+        localStorage.setItem("mentorData", encryptData(entityData));
+        setIsMentorLoggedIn(true);
+      }
+
+      router.push("/");
+    } catch (error) {
+      console.log(error);
+      if (
+        error.response &&
+        error.response.status >= 400 &&
+        error.response.status <= 500
+      ) {
+        toast.error(error.response.data.message);
+      }
+    }
+  };
+
+  async function handleCallbackResponse(response) {
+    var userObject = jwt_decode(response.credential);
+    console.log(userObject);
+    try {
+      const url = `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/mentors/gloginauth`;
```

Comment:
I think url should be modified for different enitity

Reply:
oh yeah